src/megaapi_impl.cpp

```diff
@@ -15674,6 +15685,34 @@ void MegaApiImpl::checkfile_result(handle h, error e, byte*, m_off_t, m_time_t,
     }
 }
 
+void MegaApiImpl::getbanner_result(error e)
+{
+    auto it = requestMap.find(client->restag);
+    if (it == requestMap.end())
+    {
+        return;
+    }
+
+    MegaRequestPrivate* request = it->second;
+
+    fireOnRequestFinish(request, make_unique<MegaErrorPrivate>(e));
+}
+
+void MegaApiImpl::getbanner_result(vector< tuple<int, string, string, string, string, string, string> >&& banners)
+{
+    auto it = requestMap.find(client->restag);
+    if (it == requestMap.end())
+    {
+        return;
+    }
+
+    MegaRequestPrivate* request = it->second;
```

Comment:
We should test against `request` is valid (it should not, but it could be nullptr) and against the type of request (sometimes, the same `restag` is used for multiple requests (when they are chained). Just for homogeneity, we test type for all commands.
```suggestion cpp
    if(requestMap.find(client->restag) == requestMap.end()) return;
    MegaRequestPrivate* request = requestMap.at(client->restag);
    if(!request || (request->getType() != MegaRequest::TYPE_GET_BANNERS)) return;
```